src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/GovNotifyEmailService.kt

```diff
@@ -1,11 +1,59 @@
 package uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.service
 
+import org.slf4j.LoggerFactory
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.config.Email
 import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.config.EmailService
 import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.config.EmailTemplates
 import uk.gov.service.notify.NotificationClient
 
-internal class GovNotifyEmailService(client: NotificationClient, emailTemplates: EmailTemplates) : EmailService {
-  override fun sendEmail() {
-    TODO("Service is not yet implemented")
+class GovNotifyEmailService(
+  private val client: NotificationClient,
+  private val emailTemplates: EmailTemplates,
+) : EmailService {
+
+  companion object {
+    private val log = LoggerFactory.getLogger(this::class.java)
+  }
+
+  override fun send(email: Email) {
+    when (email) {
+      is AppointmentCancelledCourtEmail -> send(email, emailTemplates.appointmentCancelledCourt)
+      is AppointmentCancelledPrisonEmail -> send(email, emailTemplates.appointmentCancelledPrison)
+      is AppointmentCancelledPrisonNoCourtEmail -> send(email, emailTemplates.appointmentCancelledPrisonNoCourt)
+      is OffenderReleasedCourtEmail -> send(email, emailTemplates.offenderReleasedCourt)
+      is OffenderReleasedPrisonEmail -> send(email, emailTemplates.offenderReleasedPrison)
+      is OffenderReleasedPrisonNoCourtEmail -> send(email, emailTemplates.offenderReleasedPrisonNoCourt)
+      is OffenderTransferredCourtEmail -> send(email, emailTemplates.offenderTransferredCourt)
+      is OffenderTransferredPrisonEmail -> send(email, emailTemplates.offenderTransferredPrison)
+      is OffenderTransferredPrisonNoCourtEmail -> send(email, emailTemplates.offenderTransferredPrisonNoCourt)
+      else -> throw RuntimeException("Unsupported email type ${email.javaClass.simpleName}.")
+    }
+  }
+
+  private fun send(email: Email, templateId: String) {
+    runCatching {
+      client.sendEmail(templateId, email.address, email.personalisation, null)
```

Comment:
This will likely need tweaking but should suffice until we start using the service.

Reply:
Yes, I think its a good idea to store the ID.